### PR TITLE
Update service_decoration to offer decoration code sample

### PR DESCRIPTION
**Pull request by matthieu88160**
This update offer a decorator code sample for the 'app.mailer' service. This allow to the newcomers to fully understand the decorator principle.

### CODE REVIEW

File: service_container/service_decoration.rst [hunk 1]
```diff
@@ -52,7 +52,33 @@ the original service is lost:
 Most of the time, that's exactly what you want to do. But sometimes,
 you might want to decorate the old one instead. In this case, the
 old service should be kept around to be able to reference it in the
-new one. This configuration replaces ``app.mailer`` with a new one, but keeps
+new one.
+
+Suppose you want to process the text parts to send behind the ``app.mailer``, you will create a mailer decorator::
+
+    // AppBundle/DecoratingMailer.php
+    namespace AppBundle\DecoratingMailer;
```
Comment (HeahDude) [inline service_container/service_decoration.rst]: To be consistent, the namespace should be only `AppBundle`.

**Review by HeahDude — COMMENTED** [review]
Thanks for opening that PR. I kind of like this change, I'm just not sure yet if we should be that explicit here.

File: service_container/service_decoration.rst (hunk 1 above)
Comment (matthieu88160) [inline service_container/service_decoration.rst]: Oh, I've dona a mistake. What about using :: AppBundle\Decorator

Comment (HeahDude) [inline service_container/service_decoration.rst]: If we were to add a namespace I think we should use `Mailer`. But then other examples would need an update too.
What do other thinks?

Comment (matthieu88160) [inline service_container/service_decoration.rst]: We must provide the better practice to follow in the documentation examples and I think adding a namespace is a good practice in this context case. I'll update to `AppBundle\Mailer` tomorrow.

**Comment by matthieu88160** [comment]
Update done for Mailer service/class namespace in service_container and DependencyInjection component

**Review by roman-1983 — APPROVED** [review]

**Comment by javiereguiluz** [comment]
@matthieu88160 I'm sorry we didn't take care of your proposal on time. During the past months, the Symfony Docs team has decided to aggressively simplify everything because we don't have enough resources to maintain everything. In the past weeks we've deleted thousands of lines, even entire articles.

About your specific changes, changing `use AppBundle\Mailer;` to `use AppBundle\Mailer\Mailer;` is not that important in the services articles (and some readers could think that the double `Mailer` is some kind of error).

About the other proposal, I don't think it's necessary to display an example of the source code of the `DecoratingMailer`. The reason is that this service is not different to any other service explained in the other docs, so this would be a content duplication. The `service_decoration.rst` article should focus only on explaining what is different (the `decorates` option, etc.)

That's why I'm closing this as "won't fix" and I hope you understand why. Thanks!

**Pull request closed without merging**